### PR TITLE
[DOCS] Update the docs & blogs for 1.5.0 release

### PR DESCRIPTION
**Pull request by mynameborat**
**Changes**: 
- Version bumps to next release
- Release blog for 1.5.0
- Other 1.5.0 documentation changes

### CODE REVIEW

**Comment by abhishekshivanna** [comment]
Thank you! LGTM

File: docs/_blog/2020-07-01-announcing-the-release-of-apache-samza--1.5.0.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+Job Runner will now simply submit Samza job to Yarn RM without executing any user code and job planning will happen on ClusterBasedJobCoordinator instead. This simplified workflow addresses security requirements where job submissions need to be isolated in order to execute user code as well as operational pain points where deployment failure could happen at multiple places.
+
+Full list of the jiras addressed in this release can be found [here](https://issues.apache.org/jira/issues/?jql=project%20%3D%20SAMZA%20and%20fixVersion%20in%20(1.5)).
+
+### Upgrading your application to Apache Samza 1.5.0
+ConfigFactory is deprecated as Job Runner does not load full job config anymore. Instead, ConfigLoaderFactory is introduced to be executed on ClusterBasedJobCoordinator to fetch full job config.
+If you are using the default PropertiesConfigFactory, simply switching to use the default PropertiesConfigLoaderFactory will work, otherwise if you are using a custom ConfigFactory, kindly creates its new counterpart following ConfigLoaderFactory. 
+
+Configs related to job submission must be explicitly provided to Job Runner as it is no longer loading full job config anymore.
```
[the remark this answers is not in the document]
Comment (kw2542) [reply docs/_blog/2020-07-01-announcing-the-release-of-apache-samza--1.5.0.md:71]: Very good question, I do have detailed instructions on previous PRs, do you think we should link to them or provide the instruction on the release blog as well?

[the remark this answers is not in the document]
Comment (cameronlee314) [reply docs/_blog/2020-07-01-announcing-the-release-of-apache-samza--1.5.0.md:71]: I'm not sure we should link to PRs as official documentation for steps to do an upgrade.
We could copy the details from the PRs to somewhere else, like the blog. @mynameborat any other ideas?

[the remark this answers is not in the document]
Comment (mynameborat) [reply docs/_blog/2020-07-01-announcing-the-release-of-apache-samza--1.5.0.md:71]: @cameronlee314 agreed. @kw2542, can you point me to those PRs and I can copy/add the texts to the blog.

**Comment by mynameborat** [comment]
Addressed all the comments. Getting this patch in to publish to docs.
We can fix forward any more details in the apache blog directly if needed.

**Pull request merged**